src/daycare.c

```diff
@@ -1056,15 +1057,21 @@ static u16 DetermineEggSpeciesAndParentSlots(struct DayCare *daycare, u8 *parent
         }
     }
 
+    motherEggSpecies = GetEggSpecies(species[parentSlots[0]]);
+    fatherEggSpecies = GetEggSpecies(species[parentSlots[1]]);
     hasMotherEverstone = ItemId_GetHoldEffect(GetBoxMonData(&daycare->mons[0].mon, MON_DATA_HELD_ITEM)) == HOLD_EFFECT_PREVENT_EVOLVE;
     hasFatherEverstone = ItemId_GetHoldEffect(GetBoxMonData(&daycare->mons[1].mon, MON_DATA_HELD_ITEM)) == HOLD_EFFECT_PREVENT_EVOLVE;
-
-    if (hasMotherEverstone)
-        parentSpecies = species[parentSlots[0]];
-    else if (hasFatherEverstone && GET_BASE_SPECIES_ID(GetEggSpecies(species[parentSlots[0]])) == GET_BASE_SPECIES_ID(GetEggSpecies(species[parentSlots[1]])))
-        parentSpecies = species[parentSlots[1]];
+    motherIsForeign = IsSpeciesForeignRegionalForm(motherEggSpecies, REGION_CURRENT);
+    fatherIsForeign = IsSpeciesForeignRegionalForm(fatherEggSpecies, REGION_CURRENT);
+
+    if (motherIsForeign && hasMotherEverstone)
```

Comment:
In case of e.g. female kantonian growlithe (everstone) and male hisuian growlithe (also everstone) this would incorrectly result in hisuian growlithe. Correct would be kantonian growlithe.

Reply:
Hmm, Bulbapedia claims this:
> If a Pokémon has [regional forms](https://bulbapedia.bulbagarden.net/wiki/Regional_form), then its offspring will always hatch into the form native to that region. However, if a parent of a foreign form is holding an [Everstone](https://bulbapedia.bulbagarden.net/wiki/Everstone) and is from the same evolutionary family as the offspring, then the offspring will hatch into that parent's form.

Since they don't specify gender, I am to assume that the case you propose would have Hisuian Growlithe as the offspring.

Reply:
If both parents hold an everstone it will inherit the mothers form